homeassistant/components/bouncie/sensor.py

```diff
@@ -0,0 +1,203 @@
+"""Bouncie Sensors."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import BouncieDataUpdateCoordinator, const
+
+ATTRIBUTION = "Data provided by Bouncie"
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Bouncie sensor entities based on a config entry."""
+    coordinator = hass.data[const.DOMAIN][config_entry.entry_id]
+    vehicles = coordinator.data["vehicles"]
+    sensor_types: list[SensorEntityDescription] = []
+    for vehicle in vehicles:
+        sensor_types.append(
```

Comment:
Better add a `SENSORS: tuple[SensorEntityDescription, ...] = (` const and add all of them at once instead of `append` all of them one by one.

Exemples:
- https://github.com/home-assistant/core/blob/dev/homeassistant/components/adguard/sensor.py
- https://github.com/home-assistant/core/blob/dev/homeassistant/components/synology_dsm/sensor.py
- https://github.com/home-assistant/core/blob/dev/homeassistant/components/freebox/const.py

Reply:
All of the examples have only one set of sensors. In my case, there are multiple sets of sensors, one for each vehicle. If there are 3 vehicles, the sensor list should include 7x3 = 21 sensors total. Below syntax doesn't work 
```python    
SENSORS: tuple[SensorEntityDescription, ...] = (
        SensorEntityDescription(
                key=f"car-odometer-{vehicle[const.VEHICLE_VIN_KEY]}",
                icon="mdi:counter",
                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Odometer",
                device_class=SensorDeviceClass.DISTANCE,
            ),
            SensorEntityDescription(
                key=f"car-odometer-{vehicle[const.VEHICLE_VIN_KEY]}",
                icon="mdi:counter",
                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Odometer",
                device_class=SensorDeviceClass.DISTANCE,
            ) for vehicle in vehicles        
    )
```
Python complains about `for`. Seems like Python only allows single tuple element generation when `for` is used. Any other way I can do this?

Reply:
SensorEntityDescription should be constant values, shared by all vehicles.
For example:
- the key should be "car-odometer" not "car-odometer-vin"
- the name should be "Odometer" not "nickname Odometer"
- etc.

Then it is in `__init__` that you add the VIN to the unique id, and for the name you just need to set `_attr_has_entity_name = True`